### DISCUSSION TITLE
Is it possible to call cs.github.com using an API token?

### DISCUSSION THREAD

**Discussion by mdelapenya** [community/community#38271] [Other Feature Feedback, Questions, & Ideas] [answered]
I gave it a try without success, but it does work passing a header with the Cookie of my account, although the cookie gets expired after 24 hours.

My intention is to query the search service to calculate usage stats for Go packages.

**Comment by jfullstackdev** [comment]
where in the API documentation the endpoints can be found ? 

I'm looking here but I think it's not 

https://docs.github.com/en/rest/search

it can give results partially like what you are receiving but if it's not yet fully implemented in the API, I doubt it can be done using API

**Reply by mdelapenya** [reply]
I'm using `https://cs.github.com/api/count?q=$QUERYSTRING&p=1`, which I obtained from the Chrome Devtools while watching a refresh of the page.

**Comment by look** [comment, marked as answer]
There is no public API for cs.github.com. Adding an API for the new code search is on our road map, so I'd like to hear more about your use case and how you plan to use the API. Thanks!

**Reply by eddumelendez** [reply]
Hi, somewhere we can track the progress of this? would be a nice addition to the github cli

**Reply by look** [reply]
We'll announce it in this discussion forum. I'll try to update all the API discussions with a link so people get notified.

**Reply by mdelapenya** [reply]
Our use case: wed'd like to get metrics on the usage of Go a package, by version, and for that we are searching across go.mod files for the package of interest. Given that there we haven't found a way to get download metrics on a Go package, using search APIs would be enough for us to find how many occurrences of the package are there in the public repos.

Besides Go packages, we want to track Java and .NET dependencies, looking them up in the proper dependencies files: pom.xml, *.gradle, *.cproj, etc

At this moment, we are forced to refresh the cookie daily

**Reply by nicolas2mey** [reply]
One of our use case is to target usages on specific topics 

For example, we would like to know which repositories in the organization triggers workflows using only `self-hosted` "tags" for ovious reasons that our user can usee in this cases windows, ubuntu runners

that's on of the use case, there are plenty others but mainly related to observability reports got org's repos

**Reply by bschilder** [reply]
In our case, we're trying to use the API to search for all R packages that exist on GitHub. We do this by searching for DESCRIPTION files with the term "Package:" in them:
`/(?-i)Package:/ path:/(?-i)^DESCRIPTION$/`

Being able to do this via the API would be incredible!
